BEP_draft_URI.mediawiki

```diff
@@ -0,0 +1,128 @@
+<pre>
+  BEP: Proposal
+  Layer: Applications
+  Title: URI Scheme
+  Author: Sir Ian <sirian@bnb48.club>
+  Comments-Summary: No comments yet.
+  Comments-URI: N/A
+  Status: Proposed
+  Type: Standards Track
+  Created: 2019/12/13
+</pre>
+
+This BEP is derived and extended from bip-0021
+
+==Abstract==
+This BEP proposes a URI scheme for making Binance Chain payments.
+
+==Motivation==
+The purpose of this URI scheme is to enable users to easily make payments by simply clicking links on webpages or scanning QR Codes.
+
+==Specification==
+
+=== General rules for handling (important!) ===
+
+Binance Chain clients MUST NOT act on URIs without getting the user's authorization.
+They SHOULD require the user to manually approve each payment individually, though in some cases they MAY allow the user to automatically make this decision.
+
+=== Operating system integration ===
+Graphical binance clients SHOULD register themselves as the handler for the "binance:" URI scheme by default, if no other handler is already registered. If there is already a registered handler, they MAY prompt the user to change it once when they first run the client.
+
+=== General Format ===
+
+Binance URIs follow the general format for URIs as set forth in RFC 3986. The path component consists of a binance address, and the query component provides additional payment options.
+
+Elements of the query component may contain characters outside the valid range. These must first be encoded according to UTF-8, and then each octet of the corresponding UTF-8 sequence must be percent-encoded as described in RFC 3986.
+
+=== ABNF grammar ===
+
+(See also [[#Simpler syntax|a simpler representation of syntax]])
+
+ binanceurn     = "binance:" binanceaddress [ "?" binanceparams ]
+ binanceaddress = *base58
+ binanceparams  = binanceparam [ "&" binanceparams ]
+ binanceparam   = [ amountparam / memoparam / assetparam]
+
+ amountparam    = "amount=" *digit [ "." *digit ]
+ memoparam     = "memo=" *qchar
```

Comment:
IMHO the memo is not enough for the merchant. An extendible metadata field (base64 encoded 1024 maximum raw bytes for different app parsing) might be helpful. Typical usage:
1. A wallet app want to protect the privacy of recipients, so they generated a fake address in URL but generated an encrypted address that is included in the metadata field. Only wallet app can decrypt this address and make the transfer
2. Merchant want add description of product i.e. "Coca-Cola 250 ml"

Reply:
Can't agree with this more.